### ISSUE TITLE
new-win-policy not working on new window implementation

### ISSUE THREAD

**Issue by dansachelarie** [nwjs/nw.js#7012]
I am trying to get new window opened from another webview but it seems that if I use the new implementation it is not working. (the code is not firing anymore, if I remove the flag it will work)

NWJS Version : 0.36.4
Operating System : Windws

The code used is bellow:

win.on("new-win-policy",(frame, url, policy)=>{
                            console.log('not firing')
                            policy.ignore();
                            go(url)
                        })

**Comment by ffanny** [comment]
I test `new-win-policy` with below code and it works well. You can give a demo to describe your problem in details.
`<a href="http://docs.nwjs.io/en/latest/References/Window/#event-new-win-policy-frame-url-policy" target='_blank'>new win</a>

var win=nw.Window.get();
win.on('new-win-policy', function(frame, url, policy) {
  console.log("new-win-policy called: " + url); 
  // do not open the window
  policy.ignore();
  // and open it in external browser
  nw.Shell.openExternal(url);
});
`

**Comment by dansachelarie** [comment]
What I am doing is when somebody click on a button called "Go", I get the url of a text box and open it on a new nwjs windo like this:

nw.Window.open(v!=null?v:serchUrl.value, {
                id: 'local_' + WIN_ID,
                width: 1200,
                height: 600,
                frame: false
            }, function(win) {
                    win.id=WIN_ID++;
                    WINDOWS.push({win:win, id: win.id} )
                    win.x=100;
                    win.y=200;
                    addtab(v!=null?v:serchUrl.value, win.id)
                    // log the width / height = 1366 / 768
                    if(typeof win !== 'undefined'){
                        win.on('loaded', function() {
                            changeTabName(win.window.document.title, win.id)
                            if(win.loaded==undefined){
                                win.focus();
                                win.loaded=true
                            }
                        });
                        win.on("new-win-policy",(frame, url, policy)=>{
                            console.log('new win fired')
                            policy.ignore();
                            go(url)
                        })
                        /*win.on('focus', function() {
                            console.log('New window is focused');
                        });*/
                    }
                }
            );

And if I go to google and do a search and open an url using CTRL my code from new-win-policy is not fired if I use the latest version of nwjs with the --enable-features=nw2. If I use an older version without the flag it works.

**Comment by ffanny** [comment]
In many websites(eg. google), the URL is not `<a  target='_blank' href=...` but `<a href=...`, which will not emit `new-win-policy` because there is not a new window is requested from the window or a child iframe.

And I test with [https://www.baidu.com/](https://www.baidu.com/) which uses `<a target='_blank' href=...`, `new-win-policy` is emitted when with nwjs-sdk-v0.36.3-linux.

**Comment by dansachelarie** [comment]
Let me rephrase my question, why if I remove the flag --enable-features=nw2 it is working, the event is fired and if I put it back it isn't. 

Are you sure it is not a problem in this new implementation?

**Comment by dansachelarie** [comment]
Also I have to mention that I am trying to open a random result of a search on google with CTRL pressed on windows, so it should be a 'new-win-policy' event fired for sure. (even if it is only an <a href=... , without target blank attribute )

**Comment by ffanny** [comment]
Would you please provide a full reproducible code demo so that I can check the problem accurately?

**Comment by dansachelarie** [comment]
I attached my test project. Please use the last version of nwjs (with and without the flag --enable-features=nw2), press on google, do a search and open a link with CTRL pressed. It will behave differently on the 2 casses.

[nw-win2.zip](https://github.com/nwjs/nw.js/files/2978082/nw-win2.zip)

**Comment by ffanny** [comment]
I can reproduce the problem with the latest version(v0.37.0). And it seems that `nw2` has the problem about open a link with CTRL pressed which can open a new tab.  Because `new-win-policy` works well with `<a target='_blank' href=...` in `nw2` mode.
Will try to figure out and fix it.

**Comment by dansachelarie** [comment]
Any idea when it will be fixed?
